ejb3/src/main/java/org/jboss/as/ejb3/deployment/DeploymentRepositoryService.java

```diff
@@ -0,0 +1,219 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2012, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.jboss.as.ejb3.deployment;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.jboss.as.ejb3.logging.EjbLogger;
+import org.jboss.msc.service.Service;
+import org.jboss.msc.service.ServiceName;
+import org.jboss.msc.service.StartContext;
+import org.jboss.msc.service.StartException;
+import org.jboss.msc.service.StopContext;
+
+
+/**
+ * Repository for information about deployed modules. This includes information on all the deployed EJB's in the module
+ *
+ * @author Stuart Douglas
+ */
+public class DeploymentRepositoryService implements DeploymentRepository, Service<DeploymentRepository> {
+
+    public static final ServiceName SERVICE_NAME = ServiceName.JBOSS.append("ee", "deploymentRepository");
+
+    /**
+     * All deployed modules. This is a copy on write map that is updated infrequently and read often.
+     */
+    private volatile Map<DeploymentModuleIdentifier, DeploymentHolder> modules;
+
+    private final List<DeploymentRepositoryListener> listeners = new ArrayList<DeploymentRepositoryListener>();
+
+    /**
+     * Keeps track of whether the repository is suspended or not
+     */
+    private volatile boolean suspended = false;
+
+
+    @Override
+    public void start(StartContext context) throws StartException {
+        modules = Collections.emptyMap();
+    }
+
+    @Override
+    public void stop(StopContext context) {
+        modules = Collections.emptyMap();
+    }
+
+    @Override
+    public DeploymentRepository getValue() throws IllegalStateException, IllegalArgumentException {
```

Comment:
You can drop these from the throws clause.  Not only are they not thrown, but they are RuntimeExceptions.